python3/packages/observer.py

```diff
@@ -288,12 +289,13 @@ def instrument_function(wrapped, _, args, kwargs):
                     # class or classmethod
                     aspan.set_attribute("xs.span.args.str", str(args))
                     aspan.set_attribute("xs.span.kwargs.str", str(kwargs))
-                else:
-                    # function, staticmethod or instancemethod
+                elif type(wrapped)==types.FunctionType or type(wrapped)==types.MethodType:
```

Comment:
```suggestion
                elif isinstance(wrapped, types.FunctionType) or isinstance(wrapped, types.MethodType):
```

Reply:
See my updated suggestion to commit this change below:
A tuple, as in `isinstance(x, (A, B))`, may be given as the target to check against. This is equivalent to `isinstance(x, A) or isinstance(x, B)`:

Reply:
I've looked into why this patch wasn't working, and it looks like PartialCallableObjectProxy is a subclass of Function, so isinstance(wrapped, types.FunctionType) returns True when we don't want it to. I've changed the logic to look specifically for PartialCallableObjectProxy instead of the other way around and it's passing on this job (4006917) and the Storage: BST (198455)